cmd/server/main.go

```diff
@@ -131,10 +130,15 @@ func buildCLI() *cli.App {
 
 				logger := log.NewZapLogger(log.BuildZapLogger(cfg.Log))
 
-				dynamicConfigClient, err := dynamicconfig.NewFileBasedClient(&cfg.DynamicConfigClient, logger, temporal.InterruptCh())
-				if err != nil {
-					logger.Info("Unable to create file based dynamic config client, use no-op config client instead.", tag.Error(err))
+				var dynamicConfigClient dynamicconfig.Client
+				if cfg.DynamicConfigClient != nil {
+					dynamicConfigClient, err = dynamicconfig.NewFileBasedClient(cfg.DynamicConfigClient, logger, temporal.InterruptCh())
+					if err != nil {
+						return cli.Exit(fmt.Sprintf("Unable to create dynamic config client. Error: %v", err), 1)
```

Comment:
This is very dangerous as it has the potential to cause outages in production where someone is already running Temporal with broken DynamicConfig.  So previously where things use to work with default config will now result in server won't even starting up.  In my opinion, we should not change the behavior and instead just improve the error message.

Reply:
This is what I wrote in "Risk" section in PR description. My rationale is following though: if default dynamic config is used, then everything will work fine, as we are sure that default dynamic config is correct. If someone already tried to modify config, made a mistake and gave up then there are bigger problem hidden behind it. And seems to me it is much worse than server crush after deployment of next release. It worth to add special section about it in 1.13 release notes, but swallowing this error can lead to unrecoverable problems.

Reply:
I like doing it in 2 steps.